### PR TITLE
fix display of read more / read less on TruncatedText component

### PR DESCRIPTION
**Pull request by aliceriot**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?

one part of #2466 

#### What's this PR do?

Currently when the `TruncatedText` component is used with the `showExpansionControls` prop we always show the 'read more' / 'read less' controls, regardless of whether any content has been hidden by the `Dotdotdot` component. This changes things so that we only show the expansion controls in the case where they can, you know, do something ^_^

In order to do this I had to figure out how to tell when the text was being truncated. Basically, this is being done with a div with `overflow: hidden` on it, and the `Dotdotdot` component doesn't have a way to pass in a callback or something to get out whether or not it's done the truncated.

So what I had to do was get a ref to the div used by `Dotdotdot` and then look at some of it's properties. Basically if the `scrollHeight`, which measures the height of the content, is higher than the `offsetHeight`, which represents the high of the element on screen, then there is some content which has been hidden. So in this case we show the expansion controls, and otherwise we can safely hide them. I had to do a very slight hack to make this work, basically to force our `TruncatedText` component to always do two render calls so that the `Dotdotdot` component has completed it's render. But things seem ok!

#### How should this be manually tested?

make sure that the behavior described above has been implemented and nothing works weirdly!

we use this in the compact post display and in the learning resource description.

basically, if the text is short you shouldn't see any controls in the learning resource description, and if it's long enough to overflow and get truncated you should see a 'read more / read less' control.

#### Screenshots (if appropriate)

showmorefffshort
showmorefff
showmore

### CODE REVIEW

File: static/js/components/TruncatedText.js [hunk 1]
```diff
@@ -29,6 +29,31 @@ export default function TruncatedText(props: Props) {
...
+      // the .container on the Dotdotdot holds the ref for the div it uses to
+      // wrap the text
+      const { container } = dotRef.current
+
+      if (container.scrollHeight === 0 && container.offsetHeight === 0) {
+        // the div which holds the text has been added to the DOM but doesn't have
+        // a calculated height yet. in this case we need to use a setState call
+        // to force the component to re-render so we can get the values we need
+        // to figure out when the div has overflow
+        forceReRender(val => val + 1)
```
Comment (mbertrand) [inline static/js/components/TruncatedText.js]: I think this loop is being called endlessly if the resource's short description is "", for example course MITx+12.340x ("Outdated - Global Warming Science") causes the following exception and the page goes blank:

Screen Shot 2019-12-16 at 12 01 42 PM

Comment (aliceriot) [reply static/js/components/TruncatedText.js]: ah, that is not good! will fix haha

**Comment by aliceriot** [comment]
@mbertrand ok that should be fixed now
